Introduce weather fetch example

1blckhrt opened the pull request:
Helps fix #981

<!-- Usage demo -->
```bash
cd examples
./fetch-weather.mjs London
zx ./fetch-weather.mjs "New York"
```

Output of example:

![image](https://github.com/user-attachments/assets/2af1daad-d5e9-4f48-b919-abbb75dedcd3)


- [ ] Tests pass
- [ ] Appropriate changes to README are included in PR

I am unsure whether I need to do tests, as this is a feature example to showcase `zx`, but not a feature for `zx` itself. Please let me know if I should do that. 

Also, please let me know if you would like more examples than just the one I provided. This is a really neat project and I'm happy to provide more.

1blckhrt commented:
The reason for the force-push is because I didn't know I needed to have a GPG key setup. I've gone ahead and done that and pushed again.

antongolub commented:
@1blckhrt ,

The concept is fine, imo, but let's use a weather service that does not require access key. Just to let user to apply the script via copy-paste w/o extra actions, registration. etc.

1blckhrt commented:
Sure thing, I'll see what I can find.

1blckhrt commented:
Apologies for the duplicate commit and force-push - I'm on a different machine and forgot to import my GPG key. I've implemented all of your suggestions and selected a different API to use as per the requirements. I've added the bash alias tutorial at the bottom of the file in comments. Please let me know if you would like it elsewhere.

antongolub reviewed (approved):
Seems good! I'm ready to accept it in its current version.

antongolub commented on examples/fetch-weather.mjs:
```diff
@@ -0,0 +1,101 @@
+    exit('❌ Missing weather data in API response.')
+  }
+
+  const location = area.areaName[0].value
```
It is better to separate data extraction and formatting.

```js
  const location = area.areaName[0].value
  const condition = current.weatherDesc[0].value
  const temperature = current.temp_C
  const humidity = current.humidity

  echo(chalk.yellow(`🌤️ Weather in ${location}: ${condition}`))
  echo(chalk.red(`🌡️ Temperature: ${temperature}°C`))
  echo(chalk.blue(`💧 Humidity: ${humidity}%`))
```

antongolub commented on examples/fetch-weather.mjs:
```diff
@@ -0,0 +1,101 @@
+    data = await spinner(
+      `📡 Fetching weather for "${city}" from wttr.in...`,
+      async () => {
+        const res = await fetch(`https://wttr.in/${encodeURIComponent(city)}?format=j1`)
```
https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout_static
```js
{
  signal: AbortSignal.timeout(5000),
}
```

1blckhrt commented:
I've implemented all of the suggestions. I really appreciate your thorough feedback.

antongolub commented:
Thanks for the contribution!
